tests/lib/prepare-restore.sh

```diff
@@ -696,10 +696,13 @@ prepare_suite_each() {
         return 0
     fi
 
+    # shellcheck source=tests/lib/prepare.sh
+    . "$TESTSLIB"/prepare.sh
+    # Each individual task may potentially set the SNAP_NO_MEMORY_LIMIT variable
+    prepare_memory_limit_override
+
     if [[ "$variant" = full ]]; then
```

Comment:
We need something like this 

    if [[ "$variant" = full ]]; then
        # shellcheck source=tests/lib/prepare.sh
        . "$TESTSLIB"/prepare.sh
        if os.query is-classic; then
            prepare_each_classic
        else
            prepare_each_core
        fi
    fi

Reply:
and in prepare_each_core we just call prepare_memory_limit_override

Reply:
done